# Update fix-daemon-install patch to use automake magic to avoid the wrapper scripts
#
# original by RP, updated to handle dbus-daemon-launch-helper by Kevin Tian <kevin.tian@intel.com>, 2010-07-10
Index: dbus-1.2.14/bus/Makefile.am
===================================================================
--- dbus-1.2.14.orig/bus/Makefile.am	2009-04-17 20:45:29.000000000 +0100
+++ dbus-1.2.14/bus/Makefile.am	2009-11-13 09:47:24.000000000 +0000
@@ -149,7 +149,7 @@ endif
 
 ## we use noinst_PROGRAMS not check_PROGRAMS so that we build 
 ## even when not doing "make check"
-noinst_PROGRAMS=$(TESTS) dbus-daemon dbus-daemon-launch-helper-test dbus-daemon-launch-helper
+noinst_PROGRAMS=$(TESTS) dbus-daemon-launch-helper-test
 
 bus_test_system_SOURCES=			\
 	$(XML_SOURCES)				\
@@ -171,34 +171,22 @@ bus_test_SOURCES=				\
 bus_test_LDADD=$(top_builddir)/dbus/libdbus-convenience.la $(DBUS_BUS_LIBS)
 bus_test_LDFLAGS=@R_DYNAMIC_LDFLAG@
 
+dbusdaemondir = $(DBUS_DAEMONDIR)
+dbusdaemon_PROGRAMS = dbus-daemon
+
+libexec_PROGRAMS = dbus-daemon-launch-helper
+
 ## mop up the gcov files
 clean-local:
 	/bin/rm *.bb *.bbg *.da *.gcov || true
 
-uninstall-hook:
-	rm -f $(DESTDIR)$(DBUS_DAEMONDIR)/dbus-daemon
-	rm -f $(DESTDIR)$(libexecdir)/dbus-daemon-launch-helper
-
 install-data-hook:
-	if test '!' -d $(DESTDIR)$(DBUS_DAEMONDIR); then \
-		$(mkinstalldirs) $(DESTDIR)$(DBUS_DAEMONDIR); \
-		chmod 755 $(DESTDIR)$(DBUS_DAEMONDIR); \
-	fi
-	$(INSTALL_PROGRAM) dbus-daemon $(DESTDIR)$(DBUS_DAEMONDIR)
 	$(mkinstalldirs) $(DESTDIR)$(localstatedir)/run/dbus
 	$(mkinstalldirs) $(DESTDIR)$(configdir)/system.d
 	$(mkinstalldirs) $(DESTDIR)$(configdir)/session.d
 	$(mkinstalldirs) $(DESTDIR)$(datadir)/dbus-1/services
 	$(mkinstalldirs) $(DESTDIR)$(datadir)/dbus-1/system-services
 	$(mkinstalldirs) $(DESTDIR)$(libexecdir)/dbus-1
-	$(INSTALL_PROGRAM) dbus-daemon-launch-helper $(DESTDIR)$(libexecdir)
-	if test `id -u` -eq 0; then \
-		chown root:$(DBUS_USER) $(DESTDIR)$(libexecdir)/dbus-daemon-launch-helper; \
-		chmod 4750 $(DESTDIR)$(libexecdir)/dbus-daemon-launch-helper; \
-	else \
-		echo "Not installing $(DESTDIR)$(libexecdir)/dbus-daemon-launch-helper binary setuid!"; \
-		echo "You'll need to manually set permissions to root:$(DBUS_USER) and permissions 4750"; \
-	fi
 
 #### Init scripts fun
 SCRIPT_IN_FILES=messagebus.in \
